src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3284,12 +3285,27 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                                 );
                                 throw SemanticAbort();
                             }
-                            break;
+                        }
+                        if (arg_name == "end") {
+                            visit_expr(*c->m_keywords[i].m_value);
+                            end = ASRUtils::EXPR(tmp);
+                            ASR::ttype_t *type = ASRUtils::expr_type(end);
+                            if (!ASRUtils::is_character(*type)) {
+                                std::string found = ASRUtils::type_to_str(type);
+                                diag.add(diag::Diagnostic(
+                                    "End is expected to be of string type",
+                                    diag::Level::Error, diag::Stage::Semantic, {
+                                        diag::Label("Expected string, found: " + found,
+                                                {separator->base.loc})
```

Comment:
```suggestion
                                                {end->base.loc})
```

This will fix the issue.

Reply:
Thanks @Smit-create . I think the tests should pass now.